bciers/apps/registration/app/components/operations/Operations.tsx

```diff
@@ -0,0 +1,74 @@
+import { GridRowsProp } from "@mui/x-data-grid";
+
+import buildQueryParams from "@/app/utils/buildQueryParams";
+import OperationDataGrid from "./OperationDataGrid";
+import { FrontEndRoles } from "@/app/utils/enums";
+import { OperationRow, OperationsSearchParams } from "./types";
+import { actionHandler } from "@/app/utils/actions";
+
+export const formatOperationRows = (rows: GridRowsProp) => {
+  if (!rows) {
+    return;
+  }
+  return rows.map(({ id, operator, name, bcghg_id, type }) => {
+    return {
+      id,
+      name,
+      bcghg_id,
+      operator: operator,
+      type,
+    };
+  });
+};
+
+// 🛠️ Function to fetch operations
+export const fetchOperationsPageData = async (
+  searchParams: OperationsSearchParams,
+) => {
+  try {
+    const queryParams = buildQueryParams(searchParams);
+    // fetch data from server
+    const pageData = await actionHandler(
+      `registration/v2/operations${queryParams}`,
+      "GET",
+      "",
+    );
+    return {
+      rows: formatOperationRows(pageData.data) as OperationRow[],
```

Comment:
`rows: formatOperationRows(pageData.data) ?? []` seems to make typescript happy without asserting the type.

Reply:
I didn't need to cast it after removing the function